src/layer_clear_selective.lua

```diff
@@ -0,0 +1,70 @@
+function plugindef()
+    finaleplugin.RequireSelection = true
+    finaleplugin.Author = "Carl Vine"
+    finaleplugin.AuthorURL = "http://carlvine.com/?cv=lua"
+    finaleplugin.Copyright = "CC0 https://creativecommons.org/publicdomain/zero/1.0/"
+    finaleplugin.Version = "v1.03"
+    finaleplugin.Date = "2022/06/09"
+    finaleplugin.CategoryTags = "Note"
+    finaleplugin.Notes = [[
+        Clear all music from the chosen layer in the surrently selected region. 
+        (Note that all of a measure's layer will be cleared even if it is partially selected).
+    ]]
+    return "Clear layer selective", "Clear layer selective", "Clear the chosen layer"
+end
+
+-- RetainLuaState will return global variable: clear_layer_number
+
+function clear_chosen_layer(layer_number)
```

Comment:
There's a library function `layers.clear`. Would that work here?

https://www.finalelua.com/docs/library/layer#clear

Reply:
Yes of course. Silly me. Amended version follows. (This is a really old script that I've dragged up to fin-lua expectations).